### PR TITLE
feat(deployment): upgrade strategy

### LINKED ISSUE #365: Customize deployment rollout strategy
https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#strategy

### PR DESCRIPTION
**Pull request by iliapolo**
Resolves https://github.com/cdk8s-team/cdk8s-plus/issues/365

### CODE REVIEW

File: src/deployment.ts [hunk 1]
```diff
@@ -115,6 +122,11 @@ export class Deployment extends Resource implements IPodTemplate {
    */
   public readonly replicas: number;
 
+  /**
+   * The upgrade startegy of this deployment.
```
Comment (Chriscbr) [inline src/deployment.ts]:
```suggestion
   * The upgrade strategy of this deployment.
```

File: test/deployment.test.ts [hunk 2]
```diff
@@ -201,3 +203,58 @@ test('Synthesizes spec lazily', () => {
...
+test('default deployment strategy', () => {
+
+  const chart = Testing.chart();
+
+  const deployment = new kplus.Deployment(chart, 'Deployment');
+  deployment.addContainer({ image: 'image' });
+
+  const spec: k8s.DeploymentSpec = Testing.synth(chart)[0].spec;
+
+  expect(deployment.strategy).toEqual(DeploymentStrategy.rollingUpdate());
```
Comment (Chriscbr) [inline test/deployment.test.ts:216]: I'm surprised this object equality works (I expected you'd have to call `_toKube()` on each?)

**Review by Chriscbr — APPROVED** [review]

**Comment by iliapolo** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|k8s-21/main|[](https://github.com/cdk8s-team/cdk8s-plus/pull/589)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Comment by iliapolo** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|k8s-20/main|[](https://github.com/cdk8s-team/cdk8s-plus/pull/590)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**